database.cc

```diff
@@ -2326,6 +2326,70 @@ future<> database::drain() {
     co_await _commitlog->shutdown();
 }
 
+static sstring extract_cf_name(const sstring& directory_name) {
+    // cf directory is of the form: 'cf_name-timestamp'
+    return directory_name.substr(0, directory_name.find("-"));
+}
+
+future<std::vector<database::snapshot_details_result>>  database::get_snapshot_details() {
+    std::vector<sstring> data_dirs = _cfg.data_file_directories();
+    auto dirs_only_entries = lister::dir_entry_types{directory_entry_type::directory};
+    std::vector<database::snapshot_details_result> details;
+
+    for (auto& datadir : data_dirs) {
+        co_await lister::scan_dir(datadir, dirs_only_entries, [this, &dirs_only_entries, &details] (fs::path parent_dir, directory_entry de) -> future<> {
```

Comment:
https://github.com/scylladb/scylla/pull/9835 can simplify this nested scan a lot

Reply:
@StarostaGit has left Scylla so I think we will probably just merge this PR. Or maybe you would like to pick it up and rebase on top on #9835 @bhalevy ?

Reply:
@haaawk #9835 can be done as a followup. it doesn't fix anything. We can proceed with this PR as is as far as I'm concerned.

@nyh any objections?

Reply:
@bhalevy @haaawk unfortunately there are new conflicts. Can one of you please rebase and fix these conflicts? I'll commit the results.

Reply:
@nyh, I'm rebasing and retesting ti.
Will submit a new PR since I can't (and don't want to) force-push into @StarostaGit's repo.